raft/log.go

```diff
@@ -959,7 +959,7 @@ func (l *Log) candidateLoop(closing <-chan struct{}) State {
 
 	// Increment term and request votes.
 	l.mu.Lock()
-	l.term++
+	l.mustSetTerm(l.term+1)
 	l.votedFor = l.id
 	term := l.term
```

Comment:
This is not actually quite right, since `l.term` is no longer being incremented.